src/librustc_passes/consts.rs

```diff
@@ -312,6 +312,10 @@ impl<'a, 'tcx> Visitor<'tcx> for CheckCrateVisitor<'a, 'tcx> {
                 Err(ConstEvalErr {
                     kind: LayoutError(ty::layout::LayoutError::Unknown(_)), ..
                 }) => {}
+                Err(ConstEvalErr { kind: Math(_), .. }) if self.tcx.sess.overflow_checks() => {}
+                // ^ Disable overflow errors since they will be emitted again in librustc_trans
+                //   so we avoid double-linting the same expression (the one in librustc_trans is
+                //   more accurate).
```

Comment:
This is not a good idea - the later codegen warning is *opportunistic* (the code has to actually get instantiated), while this is far more general.

Ideally we'd do this check on MIR and then if we *did* emit a warning from MIR then we shouldn't emit the same warning during monomorphic codegen.